### PR TITLE
Enhance kafka topic selection

### PR DESCRIPTION
**Pull request by urso**

### CODE REVIEW

File: metricbeat/metricbeat.full.yml [hunk 1]
```diff
@@ -402,14 +402,10 @@ output.elasticsearch:
   # to.
   #hosts: ["localhost:9092"]
 
-  # The Kafka topic used for produced events. If use_type is set to true, the
```
Comment (ruflin) [inline metricbeat/metricbeat.full.yml]: Do we already have some docs about how to use format string? If not, we should definitively create an issue for it.

File: libbeat/docs/outputconfig.asciidoc [hunk 2]
```diff
@@ -543,11 +543,8 @@ The cluster metadata contain the actual Kafka brokers events are published to.
 
 ===== topic
 
-The Kafka topic used for produced events. If `use_type` is set to true, the topic will not be used.
-
-===== use_type
-
-Set Kafka topic by event type. If `use_type` is false, the `topic` option must be configured. The default is false.
+The Kafka topic used for produced events. The setting can be a format string
+using any event field. To set the topic from document type use `%{[type]}`.
```
Comment (ruflin) [inline libbeat/docs/outputconfig.asciidoc]: Should type be our default recommendation? Bringing up as we had recently some discussion around type. Perhaps better use beat.name by default?

File: libbeat/outputs/kafka/config.go [hunk 3]
```diff
@@ -14,8 +14,6 @@ type kafkaConfig struct {
 	TLS             *outputs.TLSConfig `config:"tls"`
 	Timeout         time.Duration      `config:"timeout"             validate:"min=1"`
 	Worker          int                `config:"worker"              validate:"min=1"`
-	UseType         bool               `config:"use_type"`
-	Topic           string             `config:"topic"`
```
Comment (ruflin) [inline libbeat/outputs/kafka/config.go]: Topic still exist, don't we need it here?

File: libbeat/outputs/kafka/kafka_integration_test.go [hunk 4]
```diff
@@ -57,11 +59,13 @@ func newTestKafkaClient(t *testing.T, topic string) *client {
 
 func newTestKafkaOutput(t *testing.T, topic string, useType bool) outputs.Outputer {
 
+	if useType {
```
Comment (ruflin) [inline libbeat/outputs/kafka/kafka_integration_test.go]: Is this to test BC compatibility?

File: metricbeat/metricbeat.full.yml (hunk 1 above)
Comment (urso) [inline metricbeat/metricbeat.full.yml]: see #2132

File: libbeat/docs/outputconfig.asciidoc (hunk 2 above)
Comment (urso) [inline libbeat/docs/outputconfig.asciidoc]: Default recommendation should be a static string (we do not set any default and require user to set the topic name). type is commonly used with filebeat + multiple prospectors setting document_type.

File: libbeat/outputs/kafka/config.go (hunk 3 above)
Comment (urso) [inline libbeat/outputs/kafka/config.go]: Yes, Topic still exists, but not required here. `outil.BuildSelectorFrom...` looks for `topic` and `topics` building a topic-selector similar to pipelining/index selection in elasticsearch output.

File: libbeat/outputs/kafka/kafka_integration_test.go (hunk 4 above)
Comment (urso) [inline libbeat/outputs/kafka/kafka_integration_test.go]: use_type has been removed from config. This is in order to keep the tests working without much change, as some tests use the event `type` to choose topic dynamically and other tests set static topic name. Alternatively one can change tests to pass format-string in `topic` and remove `useType`.

**Comment by ruflin** [comment]
@urso needs rebase

**Pull request merged**